internal/services/storage/storage_account_resource.go

```diff
@@ -1742,10 +1749,13 @@ func resourceStorageAccountRead(d *pluginsdk.ResourceData, meta interface{}) err
 
 	if resp.Sku.Tier == storage.SkuTierStandard {
 		if resp.Kind == storage.KindStorage || resp.Kind == storage.KindStorageV2 {
-			queueClient, err := storageClient.QueuesClient(ctx, *account)
+			// Currently, only the data plane API of queues fully support the exported properties of the `queue_properties`.
+			// TODO @magodo: once the mgmt plane API catches up, switch following client to the mgmt plane.
+			queuesDataPlaneClient, err := storageClient.QueuesDataPlaneClient(ctx, *account)
 			if err != nil {
 				return fmt.Errorf("building Queues Client: %s", err)
 			}
+			queueClient := shim.NewDataPlaneStorageQueueWrapper(queuesDataPlaneClient).(shim.DataPlaneStorageQueueWrapper)
```

Comment:
ditto here, the intention of the shim layer is to avoid these implementation details within the resources, so we shouldn't be casting these inside the resources - does the `storageClient.QueuesDataPlaneClient` method not have a method defined on the interface?